### ISSUE TITLE
csv download of home page data

### ISSUE THREAD

**Issue by ultimike** [communityinclusion/escoach#100]
For home page .csv downloads, all in a single table. @thinkcollege  will let us know about adding the green arrows and blue starts as separate columns in the .csv (per our conversation on August 10).

## Important

Added Sept 13, 2023 via @thinkcollege: 

> An added feature we would like that is not on this wireframe, is an additional administrative download, for back end users with role Administrator.  Administrators should have access to a download that is similar to the TA User download, with key activities, best practices, and response statistics.  But this should be for all providers in the system, not just those assigned to a given TA user.  Additionally, Administrators should have a download of all individuals, similar to what the Employment consultant has (Role Survey participant, profile job type: Employment consultant).  That is, key activities, best practices, and response stats.  This download should include data for all Survey participant/Employment consultant users.

**Comment by ultimike** [comment]
From an email from Paul on August 10:

> Alberto wants the up arrows and stars in the CSV downloads.  We will have to figure out what those markers will look like and how to add some kind of key.

>

**Comment by ultimike** [comment]
We will add 2 columns for each metric - one for "Better than last month" and one for "Better than All".

**Comment by mgcobblestone** [comment]
@thinkcollege The admin downloads are in place now in the latest commit.  The links should appear on the Reports menu for admin users after flushing caches.

cc: @ultimike

**Comment by ultimike** [comment]
@thinkcollege I think we can close this issue for now and re-open or create new issues if we find any bugs with this functionality. Please do so if you agree.

thanks,
-mike

**Comment by thinkcollege** [comment]
@ultimike @mgcobblestone I don't see response rates or number of users in the user or admin downloads.  Also, we asked for some kind of heading or top row in the download with indication of what was in the download, like provider name, date.

**Comment by ultimike** [comment]
@thinkcollege Can you shoot me an email with the URL and credentials to login to the dev site where you pulled these changes into?

I'm seeing most of what you mentioned in your previous comment on my local, so I just want to compare.

Is this the dev site? [rp-escoach.communityinclusion.org](http://rp-escoach.communityinclusion.org/)

Also - when is the last time you did a `git pull` of our `es_home` branch to that environment?

thanks,
-mike

**Comment by thinkcollege** [comment]
@ultimike I merged es_home into devel, the branch deployed on rp-escoach.communityinclusion.org (Your email address works there as an admin, though you may have to update your password.)  I tried to sync the config but got an error message: 
    The configuration cannot be imported because it failed validation for the following reasons:
    Entities exist of type Profile and Profile type Technical Assistant. These entities need to be deleted before importing.
Do you any advice on how to get the config to work?  Would I really delete profiles and profile types?

**Comment by ultimike** [comment]
@thinkcollege Are you sure that the rp-escoach site is looking at the latest commit of the `devel` branch? What I'm seeing on rp-escoach doesn't match with what I'm seeing in the code. 

For example, on the http://rp-escoach.communityinclusion.org/admin/config/development/configuration, if you click to see the differences in the core.extension config, it indicates that the `es_homepage` module isn't enabled in "Staged" (the `config/sync/core.extension.yml` file).

I was able to ssh in using `ssh ici@104.131.35.148 -p22` (which I found in my notes from a couple of years ago), and went to `cd  /home/ici/[escoach.communityinclusion.org/escoach/web](http://escoach.communityinclusion.org/escoach/web)`, but did see the Git repository (`.git` file), so I'm a bit perplexed at what I'm looking at.

Bottom line - there should be no need to delete anything.

-mike

**Comment by thinkcollege** [comment]
The rp-escoach site is just a deployment of the devel branch.  I merged es_home into the devel branch earlier today and deployed it to the rp-escoach site.  I am also running it locally.  However, now I DO see the response rate and number of respondent figures in the individual and anonymous, and TA user downloads.  I missed it because it was crunched between the activities and best practices results.  Could you maybe insert some separation between the tables?

**Comment by thinkcollege** [comment]
Same for the admin provider download.  Could that be split into two tables?

**Comment by ultimike** [comment]
I do not believe a single .csv file can be split into multiple tables.

How about we just add some blank rows between the two tables? If this is acceptable, then we'll also add some additional metadata as you requested:

> Also, we asked for some kind of heading or top row in the download with indication of what was in the download, like provider name, date.

-mike

**Comment by thinkcollege** [comment]
Right, but you can give the illusion of more than one table with spacing and formatting.  For example, here is a spreadsheet I generate for my organization, which puts several tables into one with headers and captions
[all_states_bluebook_tables_2021.csv](https://github.com/communityinclusion/escoach/files/13273472/all_states_bluebook_tables_2021.csv)

**Comment by thinkcollege** [comment]
@ultimike  BTW I figured out the problem with the rp-escoach config.  The config/sync directory was excluded in the deployment.  It is included now.

**Comment by ultimike** [comment]
@thinkcollege I've updated the main download .csv as we discussed. Check it out and let me know if this is what you're looking for.

Also - regarding:

> An added feature we would like that is not on this wireframe, is an additional administrative download, for back end users with role Administrator. Administrators should have access to a download that is similar to the TA User download, with key activities, best practices, and response statistics. But this should be for all providers in the system, not just those assigned to a given TA user. Additionally, Administrators should have a download of all individuals, similar to what the Employment consultant has (Role Survey participant, profile job type: Employment consultant). That is, key activities, best practices, and response stats. This download should include data for all Survey participant/Employment consultant users.

These downloads are available to those with the "Administrator" role from the admin menu's "Report" dropdown. The description provided doesn't specify if these Administrator downloads should be only for the previous month, or for all-time. Let us know.

thanks,
-mike

**Comment by thinkcollege** [comment]
Let's keep it simple and make it just for the previous month.

**Comment by ultimike** [comment]
@thinkcollege I have confirmed that the 2 admin .CSVs output data from the previous month (and have added a header row to show the month and year) - committed and pushed to the `es_home` branch.

Can this issue be closed?

thanks,
-mike

**Comment by thinkcollege** [comment]
@ultimike So, it was too hard to stack the results in the admin downloads like you did for the front end downloads? It might make for easier reading.  Also I noticed that some of the headings in the admin provider and user csvs read "Better than Last All"  I think that should be "Better than all."   I am having Alberto look these over for any issues.  We will also want to do some data checking when we have it on a test site with real data loaded.

**Comment by thinkcollege** [comment]
Here are Alberto's comments about the admin downloads: 

**I reviewed the admin downloads. Overall looks great. Below are a few comments about minor details.
Both provider and user: Would it be complicated to move columns N and O (Response rate and # Respondents) to the very right side of the sheet in all reports? For TA and Participants move it to the bottom?
User report. Would it be complicated to add 4 columns for First and last name, provider's name, and unique URL of their dashboard? I am thinking about when we generate the monthly data email featuring a metric. It would save us time if we can pull from this spreadsheet their names and emails. If adding the unique links is feasible, we can include it in the email so they can go straight to their own dashboard. The URL link is a "nice to have", not must have.

Column O looks like the number of surveys completed whereas it should be the number of respondents. 
The "Better than..." columns for AFTER Hire should include a "-" to signify that there is no data (We do not flag better for this particular metric).**
Also, could the spreadsheet have a line at the bottom, something like "downloaded from https://www.es-coach.org/dashboard, 11/8/2023"

**Comment by ultimike** [comment]
@thinkcollege Parsing Alberto's requests:

1. `Both provider and user: Would it be complicated to move columns N and O (Response rate and # Respondents) to the very right side of the sheet in all reports?` - this is done.  
2. `For TA and Participants move it to the bottom?` - I think that Alberto is talking about the home page .csv download here, correct? If so, I've just made this change.  
3. `User report. Would it be complicated to add 4 columns for First and last name, provider's name, and unique URL of their dashboard? I am thinking about when we generate the monthly data email featuring a metric. It would save us time if we can pull from this spreadsheet their names and emails. If adding the unique links is feasible, we can include it in the email so they can go straight to their own dashboard. The URL link is a "nice to have", not must have.` I have added the first name, last name and provider name to the user report. For the Dashboard URL, they are all the same for all users, just `/dashboard`. I suspect what Alberto is requesting is an Auto Login URL that automatically redirects to the user's dashboard. For that, I would recommend that we defer to what we're doing in #107 
4. `Column O looks like the number of surveys completed whereas it should be the number of respondents.` I have changed it to the `totalSurveysSent` value instead of the `netResponses` value - let me know if this is what you're looking for.
5. `The "Better than..." columns for AFTER Hire should include a "-" to signify that there is no data (We do not flag better for this particular metric).**` I do not understand this request. There appears to be YES/NO data for these columns, should we ignore the data we have an replace it with a `-` for all "Better than..." fields for "AFTER hire" values? 
6. `Also, could the spreadsheet have a line at the bottom, something like "downloaded from https://www.es-coach.org/dashboard, 11/8/2023"` - added to user and provider reports. 

-mike

**Comment by thinkcollege** [comment]
1. Good
2.  That is correct
3. Correct, we don't need the auto-login URL on this spreadsheet.
4. Looks good
5. This is a problem.  See this note at the bottom of the dashboard: "Note: "Better" means more time in "Leading to hire", less time in "Administrative", less time in "Non-employment", and a higher response rate. There are no indications about time in suport "AFTER hire" because it may vary based on support needs. "
This is wrong on the web dashboard right now and is wrong in the spreadsheet too.
6. Good

**Comment by ultimike** [comment]
@mgcobblestone See item 5 in Paul's response above.

thanks,
-mike

**Comment by mgcobblestone** [comment]
@thinkcollege , I have pushed up a fix for #5. Please test and let me know if there are still any issues.

cc: @ultimike

**Comment by thinkcollege** [comment]
@mgcobblestone  When I checkout the es_home branch and use it locally, I get a config/sync error message, "Unable to install the auto_login_url module since it does not exist."  I don't see that module in contrib or custom.  There is a contrib module of that name but it doesn't appear to have a Drupal 10 version.

**Comment by mgcobblestone** [comment]
@thinkcollege Trying running composer install

**Comment by thinkcollege** [comment]
@mgcobblestone Looks okay.

**Comment by ultimike** [comment]
@thinkcollege If this is working as expected, please close this issue.

thanks,
-mike
